mmv1/products/networkservices/GrpcRoute.yaml

```diff
@@ -0,0 +1,249 @@
+            - !ruby/object:Api::Type::NestedObject
+              name: faultInjectionPolicy
+              description: |
+                The specification for fault injection introduced into traffic to test the resiliency of clients to backend service failure.
+              properties:
+                - !ruby/object:Api::Type::NestedObject
+                  name: delay
+                  description: |
+                    Specification of how client requests are delayed as part of fault injection before being sent to a destination.
+                  properties:
+                    - !ruby/object:Api::Type::String
+                      name: fixedDelay
+                      description: |
+                       Specify a fixed delay before forwarding the request.
+                    - !ruby/object:Api::Type::Integer
+                      name: percentage
+                      description: |
+                         The percentage of traffic on which delay will be injected.
+                - !ruby/object:Api::Type::NestedObject
+                  name: abort
+                  description: |
+                     Specification of how client requests are aborted as part of fault injection before being sent to a destination.
+                  properties:
+                    - !ruby/object:Api::Type::Integer
+                      name: httpStatus
+                      description: |
+                         The HTTP status code used to abort the request.
+                    - !ruby/object:Api::Type::Integer
+                      name: percentage
+                      description: |
+                         The percentage of traffic which will be aborted.
+            - !ruby/object:Api::Type::String
+              name: timeout
+              description: |
+                Specifies the timeout for selected route.
+            - !ruby/object:Api::Type::NestedObject
+              name: retryPolicy
+              description: |
+                Specifies the retry policy associated with this route.
+              properties:
+                - !ruby/object:Api::Type::Array
+                  name: retryConditions
+                  description: |
+                    Specifies one or more conditions when this retry policy applies.
+                  item_type: !ruby/object:Api::Type::Enum
+                      name: retryCondition
+                      description: Condition when this retry policy applies.
+                      values:
+                            - :connect-failure
+                            - :refused-stream
+                            - :cancelled
+                            - :deadline-exceeded
+                            - :resource-exhausted
+                            - :unavailable
+                - !ruby/object:Api::Type::Integer
+                  name: numRetries
+                  description: |
+                    Specifies the allowed number of retries.
```

Comment:
What do we expect if this value is unset? Do we need to be able to set the value of `0` explicitly?

I ask because we've had trouble with configuring retry behavior on other resources when 0 is a valid and meaningful value because by default Terraform won't send `0`.

Reply:
For this API 0 is not a valid value.